### PR TITLE
remove redundant option/result wrapping of return values

### PR DESCRIPTION
**Pull request by matthiaskrgr**
If a function always returns `Ok(something)`, we can return `something` directly and remove the corresponding error handling in the callers.
clippy::unnecessary_wraps

### CODE REVIEW

File: compiler/rustc_attr/src/builtin.rs [hunk 1]
```diff
@@ -1036,21 +1036,21 @@ pub fn allow_internal_unstable<'a>(
...
-    allow_unstable(sess, attrs, sym::allow_internal_unstable)
+    Some(allow_unstable(sess, attrs, sym::allow_internal_unstable))
 }
 
 pub fn rustc_allow_const_fn_unstable<'a>(
     sess: &'a Session,
     attrs: &'a [Attribute],
 ) -> Option<impl Iterator<Item = Symbol> + 'a> {
-    allow_unstable(sess, attrs, sym::rustc_allow_const_fn_unstable)
+    Some(allow_unstable(sess, attrs, sym::rustc_allow_const_fn_unstable))
```
Comment (SkiFire13) [inline compiler/rustc_attr/src/builtin.rs:1046]: Couldn't these two functions also not return `Option`?

File: compiler/rustc_attr/src/builtin.rs [hunk 2]
```diff
@@ -1036,21 +1036,21 @@ pub fn allow_internal_unstable<'a>(
...
     attrs: &'a [Attribute],
 ) -> Option<impl Iterator<Item = Symbol> + 'a> {
-    allow_unstable(sess, attrs, sym::allow_internal_unstable)
+    Some(allow_unstable(sess, attrs, sym::allow_internal_unstable))
 }
 
 pub fn rustc_allow_const_fn_unstable<'a>(
     sess: &'a Session,
     attrs: &'a [Attribute],
 ) -> Option<impl Iterator<Item = Symbol> + 'a> {
```
Comment (petrochenkov) [inline compiler/rustc_attr/src/builtin.rs:1045]:
```suggestion
) -> impl Iterator<Item = Symbol> + 'a {
```

File: compiler/rustc_attr/src/builtin.rs [hunk 3]
```diff
@@ -1036,21 +1036,21 @@ pub fn allow_internal_unstable<'a>(
...
```
Comment (petrochenkov) [inline compiler/rustc_attr/src/builtin.rs:1038]:
```suggestion
) -> impl Iterator<Item = Symbol> + 'a {
```

File: compiler/rustc_attr/src/builtin.rs (hunk 1 above)
Comment (matthiaskrgr) [reply compiler/rustc_attr/src/builtin.rs:1046]: Maybe, but they are public functions which is why I don't want to touch them (there's risk of breaking other stuff easily)

File: compiler/rustc_attr/src/builtin.rs (hunk 3 above)
Comment (matthiaskrgr) [reply compiler/rustc_attr/src/builtin.rs:1038]: Are you sure about this? These are `pub`lic functions.

File: compiler/rustc_attr/src/builtin.rs (hunk 1 above)
Comment (petrochenkov) [reply compiler/rustc_attr/src/builtin.rs:1046]: All the "other stuff" is also rustc and can be changed if broken.
These specific functions are also used only in couple of places.

**Pull request merged**